packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -6,32 +6,100 @@ import { some, groupBy } from 'lodash';
 /**
  * WordPress dependencies
  */
-import { CheckboxControl, Modal, Button } from '@wordpress/components';
+import {
+	CheckboxControl,
+	Button,
+	PanelBody,
+	PanelRow,
+} from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 import { useSelect, useDispatch } from '@wordpress/data';
-import { useState } from '@wordpress/element';
+import { useState, useCallback } from '@wordpress/element';
+import { close, page, layout, grid, blockDefault } from '@wordpress/icons';
+
+function EntityRecordState( { record, checked, onChange, closePanel } ) {
+	const { name, kind, title, key } = record;
+	const parentBlockId = useSelect( ( select ) => {
+		// Get entity's blocks.
+		const { blocks = [] } = select( 'core' ).getEditedEntityRecord(
+			kind,
+			name,
+			key
+		);
+		// Get parents of the entity's first block.
+		const parents = select( 'core/block-editor' ).getBlockParents(
+			blocks[ 0 ]?.clientId
+		);
+		// Return closest parent block's clientId.
+		return parents[ parents.length - 1 ];
+	}, [] );
+
+	const { selectBlock } = useDispatch( 'core/block-editor' );
+	const selectParentBlock = useCallback(
+		() => selectBlock( parentBlockId ),
+		[]
+	);
+
+	const selectAndDismiss = useCallback( () => {
+		selectBlock( parentBlockId );
+		closePanel();
+	}, [] );
 
-function EntityRecordState( { record, checked, onChange } ) {
 	return (
-		<CheckboxControl
-			label={ <strong>{ record.title || __( 'Untitled' ) }</strong> }
-			checked={ checked }
-			onChange={ onChange }
-		/>
+		<PanelRow>
+			<CheckboxControl
+				label={ <strong>{ title || __( 'Untitled' ) }</strong> }
+				checked={ checked }
+				onChange={ onChange }
+			/>
+			{ parentBlockId ? (
+				<>
+					<Button
+						onClick={ selectParentBlock }
+						className="entities-saved-states__find-entity"
+					>
+						{ __( 'Select Entity' ) }
+					</Button>
+					<Button
+						onClick={ selectAndDismiss }
+						className="entities-saved-states__find-entity-small"
+					>
+						{ __( 'Select Entity' ) }
+					</Button>
+				</>
+			) : null }
+		</PanelRow>
 	);
 }
 
-function EntityTypeList( { list, unselectedEntities, setUnselectedEntities } ) {
+function EntityTypeList( {
+	list,
+	unselectedEntities,
+	setUnselectedEntities,
+	closePanel,
+} ) {
 	const firstRecord = list[ 0 ];
 	const entity = useSelect(
 		( select ) =>
 			select( 'core' ).getEntity( firstRecord.kind, firstRecord.name ),
 		[ firstRecord.kind, firstRecord.name ]
 	);
 
+	// Set icon based on type of entity.
+	const { name } = firstRecord;
+	let icon = blockDefault;
+	if ( name === 'site' ) {
+		icon = layout;
+	} else if ( name === 'page' ) {
+		icon = page;
+	} else if ( name === 'post' || name === 'wp_template' ) {
+		icon = grid;
+	} else if ( name === 'wp_template_part' ) {
+		icon = blockDefault;
+	}
+
```

Comment:
A constant enum will scale better.

Reply:
Would that be as with a `switch` case?  My apologies if I'm not understanding.

Reply:
```js
const ENTITY_NAME_ICONS = {
	site: layout,
	page: page,
	post: grid,
	wp_template: grid,
};
const icon = ENTITY_NAME_ICONS[ name ] || blockDefault;
```

Reply:
Oh yes!  That definitely seems like the cleanest and most scalable way to define this.  Thank you!